Refactored glTF exporter into class

kcoley opened the pull request:
Example saving multiple babylon scenes into .gltf/.glb:
```javascript
var createScene = function () {
    // This creates a basic Babylon Scene object (non-mesh)
    var scene = new BABYLON.Scene(engine);
    var camera = new BABYLON.ArcRotateCamera("Camera", 0, 0, 10, new BABYLON.Vector3(0, 0, 0), scene);
    // Positions the camera overwriting alpha, beta, radius
    camera.setPosition(new BABYLON.Vector3(0, 0, -20));
    // This attaches the camera to the canvas
    camera.attachControl(canvas, true);
    // This creates a light, aiming 0,1,0 - to the sky (non-mesh)
    var light = new BABYLON.HemisphericLight("light1", new BABYLON.Vector3(0, 1, 0), scene);
    // Default intensity is 1. Let's dim the light a small amount
    light.intensity = 0.7;
    // Our built-in 'sphere' shape. Params: name, subdivs, size, scene
    var sphere = BABYLON.Mesh.CreateSphere("sphere", 6, 3, scene);
    sphere.position.y = 2;
    // Our built-in 'ground' shape. Params: name, width, depth, subdivs, scene
    var ground = BABYLON.Mesh.CreateGround("ground1", 6, 6, 2, scene);
    ground.rotate(new BABYLON.Vector3(1,0,0), 3 * Math.PI/2);
    // Generates a .gltf and .bin file
    //var gltf = BABYLON.GLTF2Export.GLTF(scene, "test");
    // Generates a .glb file

    var gltf = BABYLON.GLTF2Export.GLB(scene, "test");
    BABYLON.GLTF2Export.downloadFiles(gltf);
    return scene;
};
```

deltakosh commented:
Really good.
I think the code that helps downloading the files should be part of the class as well

deltakosh commented on serializers/src/glTF/2.0/babylon.glTFSerializer.ts:
```diff
@@ -1,40 +1,40 @@
         version: string;
     }
-    interface OGLTFScene {
+    interface GLTFScene {
```
it should be IGLTFScene

deltakosh replied:
Same for all interface obviously :)

kcoley commented:
- Renamed interfaces to IGLTF
- Added return types to functions
- Renamed class to `GLTF2Export`
- Moved file download behavior to `GLTF2Export.DownloadFiles` static method

deltakosh commented:
So sorry KC, I forgot to add: could you please separate each interfaces and class into different files?

Thanks A LOT :D

kcoley commented:
Sure thing :)

kcoley commented:
`GLTFExporter` class and private interfaces have been separated to `babylon.glTFExporter.ts`.

deltakosh commented:
I let you do the last changes we discussed offline before merging

kcoley commented:
Updated to pass in list of babylon scenes instead of meshes

deltakosh commented:
No need for a list of scene as the root container is always a scene. 

But instead a list of meshes or a scene

kcoley commented:
Updated to pass in a single babylon scene instead of a list of scenes.

kcoley commented:
I will make one more tweak to support download files from Firefox

kcoley commented:
Updated downloadFiles function so should work with at least Chrome and Firefox

kcoley commented:
Just saw the other PR #3481 .  Need to make another change to IGLTFNode to support its change

kcoley commented:
Brought in change from PR #3481 and removed `mesh.name` from glTF serializer
